src/dssweb.theme.magic/dssweb/theme/magic/browser/faceted_views/datepicker/widget.py

```diff
@@ -0,0 +1,100 @@
+import logging
+
+from Products.Archetypes.public import Schema
+from Products.Archetypes.public import StringField
+from Products.Archetypes.public import StringWidget
+from Products.Archetypes.public import SelectionWidget
+
+from eea.facetednavigation import EEAMessageFactory as _
+from eea.facetednavigation.widgets import ViewPageTemplateFile
+from eea.facetednavigation.widgets.daterange.widget import formated_time
+from eea.facetednavigation.widgets.widget import Widget as AbstractWidget
+
+
+logger = logging.getLogger('dssweb.theme.magic.datepicker')
+
+
+EditSchema = Schema((
+    StringField(
+        'index',
+        schemata="default",
+        required=True,
+        vocabulary_factory='eea.faceted.vocabularies.DateRangeCatalogIndexes',
+        widget=SelectionWidget(
+            label=_(u'Catalog index'),
+            description=_(u'Catalog index to use for search'),
+            i18n_domain="eea"
+        )
+    ),
+    StringField(
+        'default',
+        schemata="default",
+        widget=StringWidget(
+            size=25,
+            label=_(u'Default value'),
+            description=_(u'Default string to search for'),
+            i18n_domain="eea"
+        )
+    ),
+))
+
+
+def ensure_date_format(date_input):
+    """ensure that the input str date is in the format YYYY-MM-DD"""
+    date = ''
+    if date_input:
+        clean = date_input.replace('/', '-')
+        try:
+            parts = map(int, clean.split('-'))
+        except ValueError:
+            pass
```

Comment:
This is in keeping with other widgets in the EEA suite, which swallow exceptions and default to not applying filtering if the widget is in error.  The default returned here is an empty string, which will result in the query in a no-op